### ISSUE TITLE
`generic_type_name` triggers incorrectly for integer generic parameters

### ISSUE THREAD

**Issue by lukaskollmer** [realm/SwiftLint#6213]
### Bug Description

[SE-0452](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0452-integer-generic-parameters.md) introduced integer generic parameters, which are placed alongside other generic parameters, but, in contrast to "normal" (type) generic parameters, are spelled like variables, ie starting with a lowercase identifier.
SwiftLint current;y doesn

```swift
// This triggers a violation:
struct Foo<let count: Int> {}
```

Mention the command or other SwiftLint integration method that caused the issue. Include stack traces or command output.

```bash
$ swiftlint
```

### Environment

* SwiftLint version: 0.59.1 (Homebrew)
* Xcode version: Xcode 26.0 (17A5305f)
* Configuration file: default configurations

**Comment by SimplyDanny** [comment]
Well, should they be spelled like normal variables? I mean, they are still generics and therefore constants in the current scope so `let COUNT: Int` would be reasonable too, wouldn't it?

If the convention is, that types are `UpperCamelCase` and everything else is `lowerCamelCase`, then you're right.

The question would be, whether the rule should ignore generic integers entirely or suggest a certain style.

**Comment by lukaskollmer** [comment]
they consistently spell them like normal variables throughout the proposal. additionally, these parameters effectively become normal static members of the type (ie, you can access them as if they were a static property).

IMO, the normal naming rules for ordinary variables should apply.

**Comment by SimplyDanny** [comment]
Fair enough. That sounds like the `identifier_name` should handle them instead, whereas `generic_type_name` is supposed to just skip them as you suggest.

**Comment by mitre88** [comment]
Current `main` already skips `GenericParameterSyntax` nodes whose specifier is `let`, and the rule description includes the exact `struct Foo<let count: Int>` / mixed generic examples. I therefore will not open a duplicate PR. This issue appears resolved by the current implementation.
